features/step_definitions/kata.rb

```diff
@@ -1,229 +1,127 @@
-        | control-plane=controller-manager |
-      })
+Given /^I wait until sandboxed operator is ready$/ do
+  timeout = 120
+  kata_ns = "openshift-sandboxed-containers-operator"
+  expected_status = "Installed"
+  expected_state = "Succeeded"
+  operator_status = {
+    instruction: "Wait till operator is installed to all worker nodes",
+    success: false,
+  }
+  operator_status[:success] = wait_for(timeout, stats: {}) do
+    step %Q/I store master major version in the :master_version clipboard/
+    if cb.master_version == "4.9"
+      @result_status = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[4].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_status
+      @result_state = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[5].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_state
     else
-      step %Q/I switch to cluster admin pseudo user/
-      step %Q|I obtain test data file "kata/release-#{cb.master_version}/deployment.yaml"|
-      @result = user.cli_exec(:apply, f: "deployment.yaml")
-      raise "Failed to deploy kata operator" unless @result[:success]
-      project(kata_ns)
+      @result_status = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[7].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_status
+      @result_state = admin.cli_exec(:get, resource: "operators", o: "jsonpath=''{..status.components.refs[8].conditions[0].type}'", n:kata_ns)[:stdout].to_s.include? expected_state
     end
-
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
-    step %Q|I obtain test data file "kata/release-#{cb.master_version}/kataconfiguration_v1_kataconfig.yaml"|
-    @result = user.cli_exec(:apply, f: 'kataconfiguration_v1_kataconfig.yaml')
-    raise "Failed to apply kataconfig" unless @result[:success]
-    step %Q/I wait for "#{kata_config_name}" install to start/
-  else
-    logger.info("There's already an existing 'kataconfig' resuing it...")
-    project(kata_ns)
-    step %Q/I switch to cluster admin pseudo user/
-    # make sure kata-operator is running first before installing the kataconfig
-    step %Q/a pod becomes ready with labels:/, table(%{
-      | control-plane=controller-manager |
-    })
   end
-  logger.info("Using kata image: #{pod.container_specs.first.image}")
-  step %Q/I wait until number of completed kata runtime nodes match for "#{kata_config_name}"/
+  raise "Failed to install sandboxed operator" unless @result_state
+  logger.info("Sandboxed operator installation status is #{expected_state}")
+end
+
+When /^I apply #{QUOTED} in #{QUOTED} namespace$/ do |kata_config_name, kata_ns|
+  step %Q/I store master major version in the :master_version clipboard/
```

Comment:
@valtair1891 I see multiple call to this step.  You can optimize it by doing something like
```
step %Q/I store master major version in the :master_version clipboard/ unless cb.master_version